### PR TITLE
tree-sitter: init at 0.15.7

### PR DESCRIPTION
**Pull request by Profpatsch**
###### Motivation for this change

init tree-sitter cli tool.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/parsing/tree-sitter/default.nix [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  # js = fetchDist {
+  #   file = "tree-sitter.js";
+  #   sha256 = sha256Js;
+  # };
+  # wasm = fetchDist {
+  #   file = "tree-sitter.wasm";
+  #   sha256 = sha256Wasm;
+  # };
+
+  # print all the grammar names mentioned in the fetch-fixtures script
```
Comment (FRidh) [inline pkgs/development/tools/parsing/tree-sitter/default.nix]: please move, for clarity, the parts related to updating the grammar into a separate `.nix` file.

Comment (Profpatsch) [reply pkgs/development/tools/parsing/tree-sitter/default.nix]: Done, see force push.

**Comment by Profpatsch** [comment]
@GrahamcOfBorg build tree-sitter

File: pkgs/development/tools/parsing/tree-sitter/default.nix (hunk 1 above)
Comment (Profpatsch) [reply pkgs/development/tools/parsing/tree-sitter/default.nix]: … and the other push, were I actually add the new file.

**Comment by Profpatsch** [comment]
@GrahamcOfBorg build tree-sitter

**Comment by Profpatsch** [comment]
Not sure there’s official support for aarch/darwin right now, the tool compiles some C stuff, probably with totally wrong compiler arguments, tbd.

**Comment by Profpatsch** [comment]
@GrahamcOfBorg build tree-sitter

File: pkgs/development/tools/parsing/tree-sitter/0001-cli-disable-web-ui.patch [hunk 2]
```diff
@@ -0,0 +1,41 @@
+From b0862845bbeac903ebb32308cc22b691ba76ed9d Mon Sep 17 00:00:00 2001
```
Comment (timokau) [inline pkgs/development/tools/parsing/tree-sitter/0001-cli-disable-web-ui.patch]: I think this file is redundant?

**Comment by Profpatsch** [comment]
The tests always segfault in different tests, pretty sure that’s a race condition.

Building works flawlessly on NixOS, merging. We can disable the test suite conditionally should it also fail on hydra.

**Pull request merged**